resources/projects/libraries/generic_robot_window/generic.c

```diff
@@ -46,7 +47,7 @@ static int time_step = 0;
 static bool is_hidden = false;
 static double refresh_rate = 0.032;  // s
 static double last_update_time = 0;  // s
-static bool isDeviceTypeControlEnabled[WB_NODE_TOUCH_SENSOR - WB_NODE_ACCELEROMETER];
+static bool isDeviceTypeControlEnabled[WB_NODE_VACUUM_GRIPPER - WB_NODE_ACCELEROMETER + 1];
```

Comment:
Was it wrong before?

Reply:
Yes, when accessing the `isDeviceTypeControlEnabled` value for `WB_NODE_TOUCH_SENSOR` type, the index was out of bounds: size ``WB_NODE_TOUCH_SENSOR - WB_NODE_ACCELEROMETER`` and accessing index ``WB_NODE_TOUCH_SENSOR - WB_NODE_ACCELEROMETER``.